### ISSUE TITLE
vinyl engine, tuple disappears after invalid upsert

### ISSUE THREAD

**Issue by zlobspb** [tarantool/tarantool#1608]
1.7.0-1219-g359d812

```Lua
box.cfg{}
s = box.schema.create_space('test', {engine = 'vinyl', if_not_exists = true})
s:create_index('test', {type = 'tree', parts = {1, 'num', 2, 'str'}, if_not_exists = true})

s:put({1, 'test', 3, 4})
s:select()
s:upsert({1, 'test', 'failed'}, {{'=', 3, 33}, {'=', 4, nil}})
s:select()
```

first select gives:
```Lua
- - [1, 'test', 3, 4]
```

but the last one gives:
```Lua
- []
```

**Comment by kostja** [comment]
Please don't forget to add a test case to the multi-engine suite 
after you fix this.

-- 
kostja

**Comment by rtsisyk** [comment]
Fixed in 1.7.1-24-g58be1f4
